client/sftp_client/directory.py

```diff
@@ -608,6 +612,35 @@ def path_is_hidden(self, path):
 			return True
 		return False
 
+	def read_file(self, local_file_path):
+		"""
+		Used to fetch the contents of target file
+		:param str local_file_path: the absolute path to target file
+		:return: the utf-8 contents of the file
+		:rtype: str
+		"""
+		if not (local_file_path and os.path.isfile(local_file_path) and os.access(local_file_path, os.R_OK)):
+			logger.warning('cannot write to local file, or file not found')
+			raise ValueError("Cannot read file {}".format(local_file_path))
+		with open(local_file_path, 'r') as file_:
+			file_contents = file_.read()
+		return file_contents
+
+	def save_file(self, local_file_path, buffer_contents):
```

Comment:
The documentation for this method and the remote need to be the same. This includes the parameter names. Since the goal is for them to be consistent the names should be the same. Also make sure the return types for all 4 are bytes. There's no reason for these to return strings other than convenience in the current usage context.